### PR TITLE
10 post article id comments

### PR DESCRIPTION
**Pull request by Raedaw**
ticket 10 POST /api/articles/:article_id/comments  with error handling

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -217,6 +217,73 @@ describe("9. GET /api/articles/:article_id/comments", () => {
...
+      body: "Awesome post!",
+    };
+    const article_id = 3;
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .expect(201)
+      .send(newComment)
+      .then(({ body: { comment } }) => {
+        expect(comment).toBeInstanceOf(Object);
+        expect(comment).toHaveProperty("comment_id");
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: I'd want to check the comment_id is what we expect it to be, so more specific check here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -217,6 +217,73 @@ describe("9. GET /api/articles/:article_id/comments", () => {
...
+      username: "icellusedkars",
+    };
+    const article_id = 3;
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .expect(400)
+      .send(newComment)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid input");
+      });
```
Comment (Poonam-raj) [inline __tests__/app.test.js:268]: I'd also check if `username` was missing that the same error was thrown

File: __tests__/app.test.js [hunk 3]
```diff
@@ -217,6 +217,73 @@ describe("9. GET /api/articles/:article_id/comments", () => {
...
+      username: "icellusedkars",
+      body: "Awesome post!",
+    };
+    const article_id = "banana";
+    return request(app)
+      .post(`/api/articles/${article_id}/comments`)
+      .expect(400)
+      .send(newComment)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid input");
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: Message would more likely want to be `invalid article id` or something similar here

File: __tests__/app.test.js [hunk 4]
```diff
@@ -217,6 +217,73 @@ describe("9. GET /api/articles/:article_id/comments", () => {
...
+      });
+  });
+});
```
Comment (Poonam-raj) [inline __tests__/app.test.js:311]: please also test for when a username is present but not found in the database - this would be a 404 not found

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
few more tests needed but looking strong 🎉 Throw this back when amendments are made

File: error-handling.js [hunk 5]
```diff
@@ -3,17 +3,35 @@ exports.handleInvalidPaths = (req, res) => {
...
+  console.log(err);
+  if (err.code === "23503") {
+    //foreign key restraint
+    console.log("23503");
+    res.status(404).send({ msg: `article ID does not exist` }); // need to also handle situation where username does not exist
+  } else {
+    next(err);
+  }
+};
+
```
Comment (Rayhaan2k9) [inline error-handling.js:27]: Not sure you need this error handler as your error handler on line 10 should do this for you. But happy to approve anyway :)

**Review by Rayhaan2k9 — APPROVED** [review]
Just a little suggestion but happy to approve :)

**Pull request merged**